### PR TITLE
Added logging for Request/Response Body and Headers

### PR DESCRIPTION
**Pull request by BarcoMasile**
## Description
This PR adds logging of all request and responses that successfully get returned to the caller.
Exceptions are already handled and logged.

## Changes
- moved logging tests under logging folder
- implemented the logging extension for request and response logging
- added tests
- adjusted tests after newly added code

## How Has This Been Tested?
- unit tests
- manually
- e2e

## References
request-response-examples-in-CW
request-response-detail-example

## PR Quality Checklist

- [x] I added tests to new or existing code
- [ ] I removed hardcoded strings and used our `i18n` solution instead (see [here](https://github.com/aws-samples/pcluster-manager/pull/175/commits/fdc6b77987c87a26f51dbc8da5d371d95ef80601))
- [x] I made sure no sensitive info gets logged at any time in the codebase (see [here](https://cheatsheetseries.owasp.org/cheatsheets/Logging_Cheat_Sheet.html)) (e.g. no user info or details, no stacktraces, etc.)
- [ ] I checked that infrastructure/update_infrastructure.sh runs without any error
- [x] I checked that `npm run build` builds without any error
- [x] I checked that clusters are listed correctly
- [x] I checked that a new cluster can be created (config is produced and dry run passes)
- [x] I checked that login and logout work as expected

In order to increase the likelihood of your contribution being accepted, please make sure you have read both the [Contributing Guidelines](../CONTRIBUTING.md) and the [Project Guidelines](../PROJECT_GUIDELINES.md)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: awslambda/entrypoint.py [hunk 1]
```diff
@@ -38,6 +40,11 @@ def lambda_handler(event: Dict[str, Any], context) -> Dict[str, Any]:
             pcluster_manager_api = _init_flask_app()
         # Setting default region to region where lambda function is executed
         os.environ["AWS_DEFAULT_REGION"] = os.environ["AWS_REGION"]
+
+        if 'requestContext' in event and 'requestId' in event['requestContext']:
+            # set apigw requestId to allow Flask app context to use it in logging
+            set_apigw_request_id_global(event["requestContext"]["requestId"])
```
Comment (mtfranchetto) [inline awslambda/entrypoint.py:46]: question: the `context` is already forwarded to the application with https://github.com/aws-samples/pcluster-manager/blob/cffc31d946abe90dd87683441ee9d37290d5b588/awslambda/serverless_wsgi.py#L263 can we re-use this part to access the `requestId` and get rid of the `LocalProxy`?
Probably an `os.environ('serverless.context')` is sufficient

File: api/logging/http_info.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    details['path'] = request.path
+    if request.args:
+        details['params'] = request.args
+    details['type'] = 'REQUEST'
+    _logger.info(details)
+
+
+def log_response_body_and_headers(_logger, response: Response):
+    details = __get_http_info(response)
+    details['type'] = 'RESPONSE'
```
Comment (mtfranchetto) [inline api/logging/http_info.py:18]: todo: I don't think `type` is needed for both request and response. The same check can be achieved by looking at the availability of the `path` param

File: api/logging/__init__.py [hunk 3]
```diff
@@ -19,7 +21,30 @@ def parse_log_entry(_logger, entry):
...
 def push_log_entry(_logger, level, message, extra):
     """ Logs a single log entry at the specified level """
     logging_fun = getattr(_logger, level, None)
-    logging_fun(message, extra=extra)
\ No newline at end of file
+    logging_fun(message, extra=extra)
+
+
+class RequestResponseLogging:
+    def __init__(self, logger, app: Flask = None, no_log_paths=['/logs']):
```
Comment (mtfranchetto) [inline api/logging/__init__.py:32]: suggestion: `urls_deny_list` is a more familiar name for the behavior needed

Comment (BarcoMasile) [reply api/logging/__init__.py:32]: Nice, thanks. I struggled to find a decent name

File: awslambda/entrypoint.py (hunk 1 above)
Comment (BarcoMasile) [reply awslambda/entrypoint.py:46]: I'll try again, I looked into it but wasn't able to

Comment (BarcoMasile) [reply awslambda/entrypoint.py]: changed in next rev

**Review by mtfranchetto — APPROVED** [review]

**Pull request merged**